pydrive2/fs/spec.py

```diff
@@ -473,6 +473,11 @@ def find(self, path, detail=False, **kwargs):
         bucket, base = self.split_path(path)
 
         seen_paths = set()
+        cached = base in self._ids_cache["dirs"]
+        if not cached:
+            dir_ids = self._path_to_item_ids(base)
+            self._cache_path_id(base, *dir_ids)
+
         dir_ids = [self._ids_cache["ids"].copy()]
```

Comment:
sorry, coming back here to double check :) one potential race condition here is if `_cache_path_id` that is being executed in some other thread got to the point where is update the first dictionary, but not yet the second. In this case dir_ids might not contain the cache for base yet.

Reply:
With base, do you mean the root of the filesystem (aka `self.base`) or the path passed in `find` (aka `base` here)?

Reply:
I mean the local var `base` value.

Reply:
yeah, looks like we need to lock `self._cache_path_id`.

Reply:
Maybe it'd have been okay if we used `dirs` here instead of `ids`. But I went with locking in #289.